src/Button/Button.tsx

```diff
@@ -1,14 +1,23 @@
 import React from 'react';
+import { StyledButton, StyledLink, StyledTypography } from './Button.styles';
+import { ButtonProps } from './types';
 
-export interface ButtonPropTypes {
-  label: string;
-  onClick?: () => void;
-}
+export const Button = ({
+  color = 'pink.main',
+  size = 'medium',
+  variant = 'contained',
+  disabled = false,
+  href,
+  children,
+  onClick,
+}: ButtonProps) => {
+  const BaseButton = (href ? StyledLink : StyledButton) as React.ElementType;
 
-/**
-- Use a button for actions.
-- The label should always be present.
-**/
-export const Button = ({ label, onClick }: ButtonPropTypes) => {
-  return <button onClick={onClick}>{label}</button>;
+  return (
+    <BaseButton $color={color} $size={size} $variant={variant} disabled={disabled} href={href} onClick={onClick}>
```

Comment:
```suggestion
    <BaseButton $color={color} $size={size} $variant={variant} disabled={disabled} href={href} onClick={onClick} {...rest}>
```
We should pass the rest of props to allow consumers add standard html attributes.